src/scripts/modules/gradebook.js

```diff
@@ -257,17 +260,71 @@
         return students;
     }
 
+    function getPathContent(path) {
+      const pathURI = id => `https://online.theironyard.com/admin/paths/${ id }`;
+
+      return new Promise( ( res ) => {
```

Comment:
I like the second,  quite a bit easier than another layer of promisees. If I follow you